File: app/src/main/java/my/com/toru/daggertutorial/MainActivity.java
package my.com.toru.daggertutorial;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import my.com.toru.daggertutorial.di.DaggerMainActivityComponent;

public class MainActivity extends AppCompatActivity {

    @Inject Test1 test;
    @Inject Test2 test2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        DaggerMainActivityComponent.builder().build().inject(this);

        test.test();
        test2.test();
    }
}

File: kdaggertutorial/src/main/java/my/com/toru/kdaggertutorial/ui/main/MainActivity.kt
package my.com.toru.kdaggertutorial.ui.main

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import my.com.toru.kdaggertutorial.R
import my.com.toru.kdaggertutorial.app.KDaggerTutorialApp
import my.com.toru.kdaggertutorial.util.DummyUtil
import my.com.toru.kdaggertutorial.ui.detail.DetailActivity
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var testclass: DummyUtil

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        (application as KDaggerTutorialApp)
                .appComponent.mainActivityBuilder()
                .build()
                .inject(this)

        Log.w(MainActivity::class.java.simpleName, testclass.testCode())

        main_btn.setOnClickListener{
            startActivity(Intent(this@MainActivity, DetailActivity::class.java))
        }
    }
}